UI/SidePanel.cpp

```diff
@@ -3040,42 +3040,56 @@ void SidePanel::RefreshInPreRender() {
 
 void SidePanel::RefreshSystemNames() {
     TemporaryPtr<const System> system = GetSystem(s_system_id);
-    // if no system object, there is nothing to populate with.  early abort.
     if (!system)
         return;
 
     // Repopulate the system with all of the names of known systems, if it is closed.
     // If it is open do not change the system names because it runs in a seperate ModalEventPump
     // from the main UI.
-    if (!m_system_name->Dropped()) {
-        m_system_name->Clear();
+    if (m_system_name->Dropped())
+        return;
 
-        // populate droplist of system names
-        std::map<std::string, int> system_map; //alphabetize Systems here
-        for (ObjectMap::const_iterator<System> sys_it = Objects().const_begin<System>();
-             sys_it != Objects().const_end<System>(); ++sys_it)
-        {
-            if (!sys_it->Name().empty() || sys_it->ID() == s_system_id) // skip rows for systems that aren't known to this client, except the selected system
-                system_map.insert(std::make_pair(sys_it->Name(), sys_it->ID()));
-        }
-        std::vector<GG::DropDownList::Row*> rows;
-        rows.reserve(system_map.size());
-        for (std::map< std::string, int>::iterator sys_it = system_map.begin(); sys_it != system_map.end(); ++sys_it)
-        {
-            int sys_id = sys_it->second;
-            rows.push_back(new SystemRow(sys_id));
-        }
-        m_system_name->Insert(rows, false);
+    m_system_name->Clear();
+
+    //Sort the names
+
+    // The system names are manually sorted here and not automatically in
+    // the vectorized insert because the ListBox is currently never
+    // resorted, inserted or deleted so this was faster when profiled.  If
+    // the performance of the std::stable_sort used in ListBox improves to
+    // N logN when switching to C++11 then this should simply insert the
+    // entire vector into the ListBox.  If the approach switches to
+    // maintaing the list by incrementally inserting/deleting system
+    // names, then this approach should also be dropped.
+    std::map<std::string, int> system_map;
```

Comment:
given that systems can now be renamed, this needs to account for multiple systems having the same name